viewing_party/party.py

```diff
@@ -1,23 +1,200 @@
+
+    return friends_unique_movies_list
+
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    subscriptions = user_data["subscriptions"]
+    
+    friends_unique_movies_list = get_friends_unique_watched(user_data)
+
+    movie_recommendations_list = []
+
+    for movie in range(len(friends_unique_movies_list)):
+        if friends_unique_movies_list[movie]["host"] in subscriptions:
+            movie_recommendations_list.append(friends_unique_movies_list[movie])
+
+    return movie_recommendations_list
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):  
+    user_data_movies_list = get_user_data_watchlist(user_data)
+
+    friends_unique_movies_list = get_friends_unique_watched(user_data)
+    
+    genre_list = []
+    for movie in range(len(user_data_movies_list)):
+        genre = user_data_movies_list[movie]["genre"]
+        genre_list.append(genre)
+    
+    movie_recommendations_list = []
+    if len(user_data["watched"]) > 0:
+        most_watched_genre = max(set(genre_list), key = genre_list.count)
+
+        for movie in range(len(friends_unique_movies_list)):
+            if friends_unique_movies_list[movie]["genre"] == most_watched_genre:
+                movie_recommendations_list.append(friends_unique_movies_list[movie])
+
+    return movie_recommendations_list
+
+
+def get_rec_from_favorites(user_data):
+    unique_user_data_movies_list = get_unique_watched(user_data)
+    
+    favorites = user_data["favorites"]
+    
+    movie_recommendations_list = []
+    if len(user_data["watched"]) > 0:
+        for movie in range(len(unique_user_data_movies_list)):
+            if unique_user_data_movies_list[movie] in favorites:
+                movie_recommendations_list.append(unique_user_data_movies_list[movie])
+
+    return movie_recommendations_list
+
+
+# -----------------------------------------
+# ------------- HELPERS -------------------
+# -----------------------------------------
+def get_user_data_watchlist(user_data):
+    user_data_movies_list = []
+
+    for i in range(len(user_data["watched"])):
+        user_data_movies_list.append(user_data["watched"][i])
+
+    return user_data_movies_list
+
+def get_user_data_watchlist_titles(user_data):
+    user_data_movies_titles_list = []
+
+    for i in range(len(user_data["watched"])):
+        user_data_movies_titles_list.append(user_data["watched"][i]["title"])
+
+    return user_data_movies_titles_list
+
+def get_friends_watchlists(user_data):
+    friends_movie_1_list = user_data["friends"][0]["watched"]
+    friends_movie_2_list = user_data["friends"][1]["watched"]
+
+    friends_movies_list = friends_movie_1_list + friends_movie_2_list
+    return friends_movies_list

```

Comment:
This function and the function below assume that each "friends" list only has two friends in it. There are situations where there will be more than just two friends, so we need to make sure this function works for any number of friends. We can leverage nested for loops here in a pretty beautiful way to make that happen:


    friend_movies = []

    #get friends movie titles
    for watched_dict in user_data['friends']:
        for movie in watched_dict['watched']:
            if movie not in friend_movies:
                friend_movies.append(movie)

    return friend_movies

This will return a list of all the unique movies all your friends have seen no matter how many you have!